golem-test-framework/src/components/worker_executor/mod.rs

```diff
@@ -71,85 +71,103 @@ async fn wait_for_startup(host: &str, grpc_port: u16, timeout: Duration) {
-    shard_manager: Arc<dyn ShardManager + Send + Sync + 'static>,
-    worker_service: Arc<dyn WorkerService + Send + Sync + 'static>,
-    redis: Arc<dyn Redis + Send + Sync + 'static>,
-    verbosity: Level,
-) -> HashMap<String, String> {
-    EnvVarBuilder::golem_service(verbosity)
-        .with_str("ENVIRONMENT", "local")
-        .with_str("WASMTIME_BACKTRACE_DETAILS", "1")
-        .with_str("GOLEM__KEY_VALUE_STORAGE__TYPE", "Redis")
-        .with_str("GOLEM__INDEXED_STORAGE__TYPE", "KVStoreRedis")
-        .with_str(
-            "GOLEM__BLOB_STORAGE__CONFIG__ROOT",
-            "/tmp/ittest-local-object-store/golem",
-        )
-        .with_str(
-            "GOLEM__KEY_VALUE_STORAGE__CONFIG__HOST",
-            &redis.private_host(),
-        )
-        .with_str(
-            "GOLEM__KEY_VALUE_STORAGE__CONFIG__PORT",
-            &redis.private_port().to_string(),
-        )
-        .with_str("GOLEM__KEY_VALUE_STORAGE__CONFIG__PREFIX", redis.prefix())
-        .with_str("GOLEM__BLOB_STORAGE__TYPE", "LocalFileSystem")
-        .with_str(
-            "GOLEM__PUBLIC_WORKER_API__HOST",
-            &worker_service.private_host(),
-        )
-        .with(
-            "GOLEM__PUBLIC_WORKER_API__PORT",
-            worker_service.private_grpc_port().to_string(),
-        )
-        .with_str(
-            "GOLEM__PUBLIC_WORKER_API__ACCESS_TOKEN",
-            "2A354594-7A63-4091-A46B-CC58D379F677",
-        )
-        .with_str(
-            "GOLEM__COMPONENT_SERVICE__CONFIG__HOST",
-            &component_service.private_host(),
-        )
-        .with(
-            "GOLEM__COMPONENT_SERVICE__CONFIG__PORT",
-            component_service.private_grpc_port().to_string(),
-        )
-        .with_str(
-            "GOLEM__COMPONENT_SERVICE__CONFIG__ACCESS_TOKEN",
-            "2A354594-7A63-4091-A46B-CC58D379F677",
-        )
-        .with_str("GOLEM__COMPILED_COMPONENT_SERVICE__TYPE", "Enabled")
-        .with_str("GOLEM__SHARD_MANAGER_SERVICE__TYPE", "Grpc")
-        .with_str(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__HOST",
-            &shard_manager.private_host(),
-        )
-        .with(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__PORT",
-            shard_manager.private_grpc_port().to_string(),
-        )
-        .with_str(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MAX_ATTEMPTS",
-            "5",
-        )
-        .with_str(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MIN_DELAY",
-            "100ms",
-        )
-        .with_str(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MAX_DELAY",
-            "2s",
-        )
-        .with_str(
-            "GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MULTIPLIER",
-            "2",
-        )
-        .with("GOLEM__PORT", grpc_port.to_string())
-        .with("GOLEM__HTTP_PORT", http_port.to_string())
-        .build()
+#[async_trait]
+pub trait WorkerExecutorEnvVars {
+    async fn env_vars(
```

Comment:
is async needed here?

Reply:
The goal is to reimplement this method on cloud. And on cloud it has to be `async`.